src/printer.js

```diff
@@ -2206,20 +2221,32 @@ function printJSXElement(path, options, print) {
   let forcedBreak = false;
 
   // Trim trailing whitespace, recording if there was a hardline
+  let numTrailing = 0;
```

Comment:
Yes, it prints extraneous newlines otherwise. I'm not sure why; may be a bug in the code you referenced, actually. Though in this case, we rely on that bug, as this PR allows a single extra newline.